setuptools/compat/py311.py

```diff
@@ -1,12 +1,26 @@
-import sys
+from __future__ import annotations
+
 import shutil
+import sys
+from typing import TYPE_CHECKING
+from collections.abc import Callable
+from typing import Any
+
+if TYPE_CHECKING:
+    from _typeshed import StrOrBytesPath, ExcInfo
 
 
-def shutil_rmtree(path, ignore_errors=False, onexc=None):
+def shutil_rmtree(
+    path: StrOrBytesPath,
+    ignore_errors: bool = False,
+    onexc: shutil._OnExcCallback | None = None,
```

Comment:
I think we should not depend on a private member of the stdlib.

Reply:
Since it's a type alias, I can simply expand it.